packages/block-library/src/query/edit/query-inspector-controls.js

```diff
@@ -114,11 +136,16 @@ export default function QueryInspectorControls( {
 	};
 	// Handles categories and tags changes.
 	const onTermsChange = ( terms, queryProperty ) => ( newTermValues ) => {
-		const termIds = newTermValues.reduce( ( accumulator, termValue ) => {
-			const termId = termValue?.id || terms.mapByName[ termValue ]?.id;
-			if ( termId ) accumulator.push( termId );
-			return accumulator;
-		}, [] );
+		const termIds = Array.from(
+			newTermValues.reduce( ( accumulator, termValue ) => {
+				const termId = getTermIdByTermValue(
+					terms.mapByName,
+					termValue
+				);
+				if ( termId ) accumulator.add( termId );
+				return accumulator;
+			}, new Set() )
+		);
```

Comment:
Out of curiosity, I compared the performance of this "reduce to a set then cast back to array" approach with Lodash's `uniq`, and `uniq` performs better, even though the performance is more variable:

<img width="426" alt="Captura de ecrã 2021-05-14, às 12 24 16" src="https://user-images.githubusercontent.com/150562/118264458-a1b1e980-b4af-11eb-98cd-c27c37730451.png">

This is just an interesting tidbit, not something that needs to be fixed!

Reply:
Interesting indeed... I'll have to look into this a bit in general, as I'm quite fond of `reduce` 😄